src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -110,6 +110,10 @@ export class MeetingManager implements AudioVideoObserver {
 
   simulcastEnabled: boolean = false;
 
+  hasAudioPermission: boolean = false;
+
+  hasVideoPermission: boolean = false;
+
```

Comment:
I can't find any better idea to avoid this. We need to access these variables in `setupDeviceLabelTrigger` and `listAndSelectDevices` unless we modify these functions. Is it acceptable or any suggestions?

Reply:
Initialize these two variables to `false` to handle the corner case like `getUserMedia({audio:false, video: false})`, in this case, the `deviceLabelTrigger` stops before executing the code to set their values.

Reply:
Yeah, this is fine. 

Really these are `wasAudioPermissionEverApparentlyGrantedOrTheBuilderPassedUsASyntheticAudioTrack`, or `weSawAnAudioTrackLastTimeWeRanAHook`, so you might want to comment about exactly what these mean and when the values change. 

They don't really mean whether we have permissions — we only know that at the exact moment we request an input stream by calling `getUserMedia`.

It's important to remember that customers can use the SDK without *ever* calling `getUserMedia` (e.g., by supplying their own synthetic video and audio streams), or they can use the SDK and manage input streams themselves by directly calling `getUserMedia` and then choosing those streams as input devices. We should make sure that `MeetingManager` behaves appropriately in those cases.